src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -21,6 +21,7 @@ const char* uct_rdmacm_cm_ep_str(uct_rdmacm_cm_ep_t *cep, char *str,
         [ucs_ilog2(UCT_RDMACM_CM_EP_GOT_DISCONNECT)] = "got_disconnect",
         [ucs_ilog2(UCT_RDMACM_CM_EP_DISCONNECTING)]  = "disconnecting",
         [ucs_ilog2(UCT_RDMACM_CM_EP_FAILED)]         = "failed",
+        [ucs_ilog2(UCT_RDMACM_CM_EP_CONNECTED)]      = "connect_cb_invoked",
```

Comment:
isn't this same as UCT_RDMACM_CM_EP_GOT_CONNECT?
can we avoid adding extra state flag?

Reply:
`UCT_RDMACM_CM_EP_GOT_CONNECT` is set on rdmacm event
`UCT_RDMACM_CM_EP_CONNECTED` is set on CB invocation

Reply:
can we remove one of the states

Reply:
they are needed to handle out of order events delivery by rdmacm

Reply:
can you suggest some other approach? deleting from callback queue by filter in normal flow does not seem right.

Reply:
deleting from callback queue handles other case - out of order in UCP produced by out of order execution by callback queue

Reply:
if there is no better solution in UCP, the proper way is add ordering support to CBQ